src/bindings/python/src/pyopenvino/core/tensor.cpp

```diff
@@ -22,6 +22,7 @@ void regclass_Tensor(py::module m) {
             }),
             py::arg("array"),
             py::arg("shared_memory") = false,
+            py::keep_alive<1, 2>(),
```

Comment:
`shared_memory` arg should be removed

Reply:
actually we do need `shared_memory` flag, for example for string Tensors -- in such case Tensor may be created only as a copy.
I introduced custom conditional keep_alive which doesn't tie arrays lifetime to Tensor if shared_memory=False

Reply:
```C++
cls.def(py::init([](py::array& array, const std::vector<size_t> shape, const ov::element::Type& ov_type) {
                return Common::tensor_from_pointer(array, shape, ov_type);
            }))
```
works just like as `shared_memory` was true, but the default value for
```C++
cls.def(py::init([](py::array& array, bool shared_memory) {
                return Common::object_from_data<ov::Tensor>(array, shared_memory);
            }))
```
is `false`.
Switch to `true` to aligning the defaults.

> string Tensors -- in such case Tensor may be created only as a copy.

It's worth describing it in docstrings that `shared_memory=true` raises an error (or ignores the flag) in that case.

Reply:
@Wovchena updated docstring and default value